slimta/edge/wsgi.py

```diff
@@ -89,13 +90,25 @@ class WsgiResponse(Exception):
 
     """
 
-    def __init__(self, status, headers=None, data=None):
+    def __init__(self, status, headers=[], data=[]):
         super(WsgiResponse, self).__init__(status)
         self.status = status
-        self.headers = headers or []
+        self.headers = [(_unicode_to_bytes(k), _unicode_to_bytes(v))
+                        for k, v in headers]
         self.data = data or []
 
 
+def _unicode_to_bytes(s):
+    """Encodes headers to iso-8859-1 bytes
+
+    https://www.python.org/dev/peps/pep-0333/#unicode-issues
+    """
+    assert isinstance(s, six.string_types), (
+        "{} is supposed to be an unicode string, not {}".format(type(s))
+    )
```

Comment:
Is this assertion necessary? In the style of duck-typing, you could just allow the call to `s.encode()` below and if it throws an exception then it definitely wasn't a string type. Unless I'm missing an edge case.